enterprise/backend/src/metabase_enterprise/sandbox/models/params/field_values.clj

```diff
@@ -56,5 +22,26 @@
   :feature :sandboxes
   [field]
   (if (field-is-sandboxed? field)
-    (get-or-create-sandboxed-field-values! field api/*current-user-id* @api/*current-user-permissions-set*)
+    (params.field-values/get-or-create-advanced-field-values! :sandbox field)
     (params.field-values/default-get-or-create-field-values-for-current-user! field)))
+
+(defenterprise hash-key-for-linked-filters
+  "Returns a hash-key for linked-filter FieldValues if the field is sandboxed, otherwise fallback to the OSS impl."
+  :feature :sandboxes
+  [field-id constraints]
+  (if (field-is-sandboxed? (db/select-one Field :id field-id))
+    (str (hash [api/*current-user-id*
+                (hash @api/*current-user-permissions-set*)
+                field-id
+                constraints]))
+    (field-values/default-hash-key-for-linked-filters field-id constraints)))
```

Comment:
> The issue currently is that it would then try to dispatch again to the EE implementation.

yes, I did that at first, but then it went into an infinite loop where it keeps calling the ee version.